lib/xapi-stdext-date/date.ml

```diff
@@ -90,4 +100,14 @@ let assert_utc _ = ()
 
 let never = of_float 0.0
 
-let eq = Ptime.equal
+let eq x y = match (x, y) with
+  | UTC x, UTC y -> Ptime.equal x y
+  | Localtime x, Localtime y -> x = y
+  | _ -> false
+
+let of_string_localtime x = Localtime x
```

Comment:
Instead of `of_string_localtime`, can we have a function that constructs a Date from the current local time? That would avoid mixing Unix and Ptime.
Then once it knows the time, and the local offset it can take a decision at to_string time how to print it : the standard way Ptime does, or the backwards compatible way (which I assume is without Z?)

Lets have a look how this API result looked like on Havana.